apps/axistream/src/axi_stream_reader.hh

```diff
@@ -0,0 +1,168 @@
+#pragma once
+
+#include <type_traits>
+#include "hls_macros.hh"
+#include "util.hh"
+#include "axi_stream_common.hh"
+#include "message.hh"
+
+/*
+ * Reads arbitrary length data (in bytes) from an AXI stream.
+ * This adheres to the Xilinx AXI Reference Guide UG 761:
+ * https://www.xilinx.com/support/documentation/ip_documentation/ug761_axi_reference_guide.pdf
+ *
+ * This standard differs from the ARM standard
+ * (https://static.docs.arm.com/ihi0051/a/IHI0051A_amba4_axi4_stream_v1_0_protocol_spec.pdf)
+ * in the following key ways:
+ * TKEEP: is ignored, since null bytes are only used in a packet with TLAST set, and the user of this API requests a packet by length.
```

Comment:
I expect that many practical packet formats are not of known size.  I think handling Keep and Last correctly is important.  Are all the formats you need fixed-size packets?